php/ClassInfo/ComposerClassmapFileRepository.php

```diff
@@ -104,13 +100,18 @@ public function find($path_pattern, ClassFilter $filter = null)
      */
     private function isValid($classpath)
     {
-        $filePath = $this->classLoader->findFile($classpath);
```

Comment:
This change is regression. 

- It allows to get fatal errors that are impossible to catch because of implementation of composer autoloader. If this autoloader found file by its findFile(), then it attempts to include it. Inclusion of class with non existing superclass cause fatal error. Therefore I decided to validate file after finding it although it is slow operation. In eb9a050 (without touching your changes) I attached unit test that show such fatal error. Please run
```
phpunit tests/php/ClassInfoRepository/ComposerClassmapFileRepositoryTest.php
```
I would like to write an extended version of this test in the original branch.

- You added new hidden dependency instead of using `$this->classInfoFactory->createClassInfo($classpath)` that would be easier to mock. Your proposition forces to use concrete implementation of `ReflectionClass`  that may be undesirable for lots of tests. 

This repository is implementation that explicitly relies on implementation of composer ClassLoader. Probably `$project_root` is unneeded in the constructor since `$classLoader` has `getClassMap()`.